File: awpak-ai/src/domain/context_mut/context_mut.rs
use serde::{Deserialize, Serialize};

use crate::domain::data::data::{DataComparator, DataFrom, DataToContext};


#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ContextMut
{
    pub from : DataFrom,
    pub to : DataToContext,
    pub condition : DataComparator
}

File: awpak-ai-builder/src/model/context_mut.ts
import { FromStatic, type DataFrom, type DataToContext } from "./data";
import { DataComparatorTrue, type DataComparator } from "./data_comparator";

export class ContextMut
{
    from : DataFrom = new FromStatic();
    to : DataToContext | undefined;
    condition : DataComparator = new DataComparatorTrue();
}